static/app/components/replays/videoReplayer.tsx

```diff
@@ -247,12 +255,16 @@ export class VideoReplayer {
     this._isPlaying = true;
     this._timer.start(videoOffsetMs);
 
-    // This is used when a replay clip is restarted
+    // This is used when a replay is restarted
     // Add another stop notification so the timer doesn't run over
     if (this._clipDuration) {
       this._timer.addNotificationAtTime(this._clipDuration, () => {
         this.stopReplay();
       });
+    } else {
+      this._timer.addNotificationAtTime(this._durationMs, () => {
+        this.stopReplay();
+      });
```

Comment:
You can move the if/else logic into the addNotificationAtTime argument to save some duplication/nesting